### ISSUE TITLE
subrules incompatible with aspects

### ISSUE THREAD

**Issue by matts1** [bazelbuild/bazel#20782]
### Description of the bug:

When you attempt to use an aspect on a subrule, bazel complains that you failed to call super in your subrule.

### Which category does this issue belong to?

Rules API

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

I've provided a minimal example at https://github.com/matts1/bazel-base/compare/subrule_demo

```
$ bazel build //:foo_child
<succeeds>
$ bazel build //:foo_child --aspects=//:demo.bzl%empty_aspect
INFO: Invocation ID: b872abcb-b027-4cfb-99be-ef1e2b721a25
ERROR: /usr/local/google/home/msta/bazel_base/BUILD.bazel:7:10: in //:demo.bzl%empty_aspect aspect on foo_child rule //:foo_child:
'super' was not called.
INFO: Analyzed target //:foo_child (0 packages loaded, 1 target configured).
WARNING: errors encountered while analyzing target '//:foo_child', it will not be built.
Analysis of target '//:foo_child' failed
INFO: Found 1 target...
Target //:foo_child up-to-date:
  bazel-bin/foo_child
ERROR: command succeeded, but not all targets were analyzed
INFO: Elapsed time: 0.103s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
 ```

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

7.0.0

**Comment by matts1** [comment]
I believe @comius is the correct person for this.

**Comment by comius** [comment]
This issue was fixed at head (it's only present in Bazel 7.0.0) release.

**Comment by Wyverald** [comment]
> This issue was fixed at head (it's only present in Bazel 7.0.0) release.

We should probably cherry-pick the fix into 7.1.0 then. Sounds good?

**Comment by brentleyjones** [comment]
Fixed with 048f048c75b89d441649a3694a419c90c66782fb.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.1.0 RC1](https://releases.bazel.build/7.1.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
